server/src/main/java/io/druid/segment/realtime/appenderator/AppenderatorImpl.java

```diff
@@ -635,7 +635,8 @@ private DataSegment mergeAndPush(final SegmentIdentifier identifier, final Sink
       final DataSegment segment = RetryUtils.retry(
           () -> dataSegmentPusher.push(
               mergedFile,
-              sink.getSegment().withDimensions(IndexMerger.getMergedDimensionsFromQueryableIndexes(indexes))
+              sink.getSegment().withDimensions(IndexMerger.getMergedDimensionsFromQueryableIndexes(indexes)),
+              true
```

Comment:
This deserves a comment about why it is `true`.

Reply:
actually it would be nicer if option came from the caller as eventually it should get used by others too and not just local indexing and kafka indexing.

Reply:
I agree, although I think it's not necessary for this PR as it may involve more substantial refactoring, and here we just want to fix the bug in #5161. But someone might want to do it one day.

Reply:
that sound alright, it would be good however to leave a comment in the javadocs of this class to remind devs of this special handling and caution that it needs to be modified if `AppenderatorImpl` is used in other cases in future.

Reply:
sounds good, will add a comment to the javadoc